### PR TITLE
[WIP] Add section about CSI Driver installation

### PR DESCRIPTION
**Pull request by davidz627**
/assign @saad-ali 
/cc @msau42 

Added a section about how to install CSI Drivers when Migration is turned on by default. There is both options for manual and automatic installation and I would recommend a hybrid approach.

### CODE REVIEW

File: contributors/design-proposals/storage/csi-migration.md [hunk 1]
```diff
@@ -321,6 +323,88 @@ with minimal disruption to users would be:
...
+### Non-Cloud and Missing Drivers
+For non-cloud and other miscellaneous missing drivers the question of
+installation becomes more nuanced. We have to strike a balance between
+unexpected resource consumption (installing everything would be unnecessary and
+wasteful) and user grief (in-tree drivers no longer work because CSI Drivers are
+not installed). There are a couple options we could take here:
+
+#### Automatic Installation
+One option is to create a tool that installs CSI Drivers automatically based on
+storage classes present. Since the storage classes for in-tree plugins have
```
Comment (jsafrane) [inline contributors/design-proposals/storage/csi-migration.md]: Storage classes are optional, scanning storage classes *and* PVs is more reliable.

**Comment by jsafrane** [comment]
I see also other option: we can install *all* drivers during migration by default. How many non-cloud drivers will be there? Gluster, NFS, iSCSI, FC, CephFS, Ceph RBD, Local, maybe Flex. I am not so sure about the rest. User / Kubernetes installer should have a way how to limit the drivers installed, but the default unattained migration could IMO install everything we have. The old cluster knew all the volume plugins, so should the new one.

For new installations, IMO we can put some burden on the installation tool, where user must specify the drivers to install. It should be easy to add a new driver later, if user chooses so. Question is, how do we tell all the Kubernetes installers around that this is a new requirement.

I personally do not like any automatic installation of a driver when user creates a StorageClass or PV. It gets too "magic" and it's hard to the user to see why something is not working.

**Comment by davidz627** [comment]
addressed comments @krzyzacy

**Comment by krzyzacy** [comment]
😂 @davidz627 you mean @jsafrane ^^ ?

**Comment by davidz627** [comment]
lol I was looking at the wrong PR my bad

**Comment by justaugustus** [comment]
Could we instead migrate this into a KEP and land it in [k/enhancements](https://github.com/kubernetes/enhancements)?

**Pull request closed without merging**